static/shared/js/typeahead.js

```diff
@@ -153,5 +153,20 @@ export function sort_emojis(objs, query) {
 
     const triage_results = triage(query, others, (x) => x.emoji_name);
 
-    return [...popular_emoji_matches, ...triage_results.matches, ...triage_results.rest];
+    const sorted_results_with_possible_duplicates = [
+        ...popular_emoji_matches,
+        ...triage_results.matches,
+        ...triage_results.rest,
+    ];
+    // remove unicode emojis with same code but different names
+    const sorted_unique_results = sorted_results_with_possible_duplicates.filter(
+        (emoji, index) =>
+            !is_unicode_emoji(emoji) ||
+            index ===
+                sorted_results_with_possible_duplicates.findIndex(
+                    (emoji_to_compare) => emoji_to_compare.emoji_code === emoji.emoji_code,
+                ),
+    );
```

Comment:
I think this is a quadratic algorithm (a loop which calls `findIndex`, which is another loop), which with thousands of emoji might be expensive. 

I think the filter could be done in linear time by just looping through the results, maintaining a set of unicode emoji codes that've been used already, and skipping entries that already exist. Not sure if there's a fancier way to do it.